plugins/inputs/passenger/passenger.go

```diff
@@ -0,0 +1,217 @@
+	# Plugin gather metric via parsing XML output of passenger-status
+	# More information about the tool:
+	#   https://www.phusionpassenger.com/library/admin/apache/overall_status_report.html
+	#
+	#
+	# If no path is specified, then the plugin simply execute passenger-status
+	# hopefully it can be found in your PATH
+	command = "passenger-status -v --show=xml"
+`
+
+func (r *passenger) SampleConfig() string {
+	return sampleConfig
+}
+
+func (r *passenger) Description() string {
+	return "Read metrics of passenger using passenger-status"
+}
+
+func (g *passenger) Gather(acc inputs.Accumulator) error {
+	if g.Command == "" {
+		g.Command = "passenger-status -v --show=xml"
+	}
+
+	out, err := exec.Command(g.Command).Output()
+
+	if err != nil {
+		return err
+	}
+
+	if err = importMetric(out, acc); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func importMetric(stat []byte, acc inputs.Accumulator) error {
+	var p info
+
+	decoder := xml.NewDecoder(bytes.NewReader(stat))
+	decoder.CharsetReader = charset.NewReaderLabel
+	if err := decoder.Decode(&p); err != nil {
+		return fmt.Errorf("Cannot parse input with error: %v\n", err)
+	}
+
+	tags := map[string]string{
+		"passenger_version": p.Passenger_version,
+	}
+	fields := map[string]interface{}{
+		"process_count":      p.Process_count,
+		"max":                p.Max,
+		"capacity_used":      p.Capacity_used,
+		"get_wait_list_size": p.Get_wait_list_size,
+	}
+	acc.AddFields("passenger", fields, tags)
+
+	for _, sg := range p.Supergroups.Supergroup {
+		tags := map[string]string{
+			"name": sg.Name,
+		}
+		fields := map[string]interface{}{
+			"get_wait_list_size": sg.Get_wait_list_size,
+			"capacity_used":      sg.Capacity_used,
+		}
+		acc.AddFields("passenger", fields, tags)
+
+		for _, group := range sg.Group {
+			for _, process := range group.Processes.Process {
+				tags := map[string]string{
+					"pid":              fmt.Sprintf("%d", process.Pid),
+					"code_revision":    process.Code_revision,
+					"life_status":      process.Life_status,
+					"process_group_id": process.Process_group_id,
+				}
+				fields := map[string]interface{}{
+					"concurrency":           process.Concurrency,
+					"sessions":              process.Sessions,
+					"busyness":              process.Busyness,
+					"processed":             process.Processed,
+					"spawner_creation_time": process.Spawner_creation_time,
+					"spawn_start_time":      process.Spawn_start_time,
+					"spawn_end_time":        process.Spawn_end_time,
+					"last_used":             process.Last_used,
+					"uptime":                process.getUptime(),
+					"cpu":                   process.Cpu,
+					"rss":                   process.Rss,
+					"pss":                   process.Pss,
+					"private_dirty":         process.Private_dirty,
+					"swap":                  process.Swap,
+					"real_memory":           process.Real_memory,
+					"vmsize":                process.Vmsize,
+				}
+				acc.AddFields("passenger", fields, tags)
```

Comment:
maybe call this `passenger_group` so it's more clear that this measurement represents a single passenger application group (if I'm understanding this correctly)